crates/holochain/tests/dht_arc.rs

```diff
@@ -0,0 +1,232 @@
+use holochain::test_utils::sweetest::SweetAgents;
+use holochain::test_utils::sweetest::SweetConductor;
+use holochain_keystore::KeystoreSender;
+use holochain_p2p::dht_arc::MAX_HALF_LENGTH;
+use holochain_p2p::dht_arc::MIN_PEERS;
+use holochain_p2p::dht_arc::MIN_REDUNDANCY;
+use kitsune_p2p::dht_arc::DhtArc;
+use kitsune_p2p::dht_arc::DhtArcBucket;
+use kitsune_p2p::*;
+use kitsune_p2p_types::dht_arc::check_for_gaps;
+use kitsune_p2p_types::dht_arc::check_redundancy;
+
+async fn get_peers(num: usize, half_lens: &[u32], keystore: KeystoreSender) -> Vec<DhtArc> {
+    let mut hl = half_lens.iter();
+    let mut iter = std::iter::repeat_with(|| hl.next().unwrap_or(&half_lens[0]));
+    let mut out = Vec::with_capacity(num);
+
+    let agents = SweetAgents::get(keystore, num).await;
+    for agent in agents {
+        let agent = holochain_p2p::agent_holo_to_kit(agent);
+        let arc = DhtArc::new(agent.get_loc(), *iter.next().unwrap());
+        out.push(arc);
+    }
+    out
+}
+
+#[tokio::test(flavor = "multi_thread")]
+#[ignore = "Too unstable"]
+// Unfortunately I can't get nodes to "settle".
+// As a node shrinks their arc they may drop below the redundancy
+// target. This is because all nodes are changing their arcs concurrently.
+// TODO: Figure out some way to stabilize convergence so a network can become
+// stable given no new nodes.
```

Comment:
I think it's because there's multiple moving arcs and as they move others arcs suddenly find themselves below the minimum shard size and boost up their coverage but it's a hard function to dampen. Will be nice to experiment with other ideas.